crates/autopilot/src/infra/blockchain/circuit_breaker.rs

```diff
@@ -0,0 +1,86 @@
+use {
+    crate::{
+        domain,
+        infra::blockchain::{contracts::deployment_address, ChainId},
+    },
+    ethcontract::{dyns::DynWeb3, transaction::TransactionResult},
+};
+
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub struct CircuitBreaker {
+    /// The authenticator contract used for allow-listing solvers to settle.
+    authenticator: contracts::GPv2AllowListAuthentication,
+    /// The safe module that is used to provide special role to EOA.
+    authenticator_role: contracts::Roles,
+    /// The EOA that is allowed to add/remove solvers.
+    authenticator_eoa: ethcontract::Account,
+}
+
+impl CircuitBreaker {
+    /// Spawns a new task in order to monitor the on-chain settlement from the
+    /// solvers
+    pub async fn new(
+        web3: DynWeb3,
+        chain: ChainId,
+        settlement: contracts::GPv2Settlement,
+        authenticator_eoa: ethcontract::Account,
+    ) -> Self {
```

Comment:
I still don't get why is putting this into a separate file better than putting it into blockchain::contracts. I don't see we gained anything with it, and would love to hear reasons.

Reply:
Well, there are several reasons for that:
- Having many fields within one struct specially when those fields don't relate necessarily with each other adds complexity and redundancy to the code
- Having a separate file just for the circuit breaker encapsulates and decouples the logic for it
- By having in the same file, and what's more, being an optional field, we need to unwrap it every time the removeSolver is called
- The circuit breaker object would be only instantiated if needed, they wouldn't be populating fields of other structs as `None`.

Reply:
Resolving this comment as I think we agreed not do it this way but rather have everything under circuit breaker.